OrbitQt/main.cpp

```diff
@@ -99,34 +99,33 @@ static outcome::result<void> RunUiInstance(
   OUTCOME_TRY(result, [&]() -> outcome::result<std::tuple<GrpcPort, QString>> {
     const GrpcPort remote_ports{/*.grpc_port =*/absl::GetFlag(FLAGS_grpc_port)};
 
-    if (deployment_configuration) {
-      OrbitStartupWindow sw{};
-      OUTCOME_TRY(result, sw.Run<OrbitSsh::Credentials>());
-
-      if (std::holds_alternative<OrbitSsh::Credentials>(result)) {
-        // The user chose a remote profiling target.
-        OUTCOME_TRY(
-            tunnel_ports,
-            DeployOrbitService(service_deploy_manager,
-                               deployment_configuration.value(), context,
-                               std::get<SshCredentials>(result), remote_ports));
-        return std::make_tuple(tunnel_ports, QString{});
-      } else {
-        // The user chose to open a capture.
-        return std::make_tuple(remote_ports, std::get<QString>(result));
-      }
-    } else {
+    if (!deployment_configuration) {
       // When the local flag is present
       return std::make_tuple(remote_ports, QString{});
     }
+
+    OrbitStartupWindow sw{};
+    OUTCOME_TRY(result, sw.Run<OrbitSsh::Credentials>());
+
+    if (!std::holds_alternative<OrbitSsh::Credentials>(result)) {
+      // The user chose to open a capture.
+      return std::make_tuple(remote_ports, std::get<QString>(result));
+    }
+
+    // The user chose a remote profiling target.
+    OUTCOME_TRY(tunnel_ports,
+                DeployOrbitService(
+                    service_deploy_manager, deployment_configuration.value(),
+                    context, std::get<SshCredentials>(result), remote_ports));
+    return std::make_tuple(tunnel_ports, QString{});
   }());
   const auto& [ports, capture_path] = result;
 
   ApplicationOptions options;
   options.grpc_server_address =
       absl::StrFormat("127.0.0.1:%d", ports.grpc_port);
 
-  OrbitMainWindow w(app, std::move(options));
+  OrbitMainWindow w(app, std::move(options), &service_deploy_manager.value());
```

Comment:
Is that a problem in the load-capture-case? `service_deploy_manager` is not set in that case, isn't it?